addons/binding/org.openhab.binding.yamahareceiver/src/main/java/org/openhab/binding/yamahareceiver/internal/protocol/xml/InputWithNavigationControlXML.java

```diff
@@ -0,0 +1,361 @@
+        while (true) {
+            String response = com.sendReceive(wrInput("<List_Info>GetParam</List_Info>"));
+            doc = XMLUtils.xml(response);
+            if (doc.getFirstChild() == null) {
+                throw new ReceivedMessageParseException("<Play_Control>GetParam failed: " + response);
+            }
+
+            currentMenu = XMLUtils.getNode(doc.getFirstChild(), "List_Info");
+
+            if (currentMenu == null) {
+                throw new ReceivedMessageParseException("<List_Info>: GetParam response invalid!");
+            }
+
+            Node nodeMenuState = XMLUtils.getNode(currentMenu, "Menu_Status");
+
+            if (nodeMenuState == null || nodeMenuState.getTextContent().equals("Ready")) {
+                break;
+            }
+
+            totalWaitingTime += YamahaReceiverBindingConstants.MENU_RETRY_DELAY;
+            if (totalWaitingTime > YamahaReceiverBindingConstants.MENU_MAX_WAITING_TIME) {
+                throw new IOException(
+                        "Menu still not ready after " + YamahaReceiverBindingConstants.MENU_MAX_WAITING_TIME + "ms");
+            }
+
+            try {
+                Thread.sleep(YamahaReceiverBindingConstants.MENU_RETRY_DELAY);
+            } catch (InterruptedException e) {
+                Thread.currentThread().interrupt();
+                throw new RuntimeException(e);
+            }
+        }
+
+        state.clearItems();
+
+        Node node = XMLUtils.getNode(currentMenu, "Menu_Name");
+        if (node == null) {
+            throw new ReceivedMessageParseException("Menu_Name child in parent node missing!");
+        }
+
+        state.menuName = node.getTextContent();
+
+        node = XMLUtils.getNode(currentMenu, "Menu_Layer");
+        if (node == null) {
+            throw new ReceivedMessageParseException("Menu_Layer child in parent node missing!");
+        }
+
+        state.menuLayer = Integer.parseInt(node.getTextContent()) - 1;
+
+        node = XMLUtils.getNode(currentMenu, "Cursor_Position/Current_Line");
+        if (node == null) {
+            throw new ReceivedMessageParseException("Cursor_Position/Current_Line child in parent node missing!");
+        }
+
+        int currentLine = Integer.parseInt(node.getTextContent());
+        state.currentLine = currentLine;
+
+        node = XMLUtils.getNode(currentMenu, "Cursor_Position/Max_Line");
+        if (node == null) {
+            throw new ReceivedMessageParseException("Cursor_Position/Max_Line child in parent node missing!");
+        }
+
+        int maxLines = Integer.parseInt(node.getTextContent());
+        state.maxLine = maxLines;
+
+        for (int i = 1; i < 8; ++i) {
+            node = XMLUtils.getNode(currentMenu, "Current_List/Line_" + i + "/Txt");
+            state.items[i - 1] = node != null ? node.getTextContent() : null;
+        }
+
+        if (observer != null) {
+            observer.navigationUpdated(state);
+        }
```

Comment:
Can we do this in another PR? Navigation does not work for the current Spotify implementation (Yamaha once again invented completely different commands for another subsystem) and to add that support later, we would need to touch this class anyway.